process.py

```diff
@@ -385,12 +386,22 @@ def get_provider_code(directory_name=None):
 
 
 def main():
+    dirname = "static/data/lms"
     parser = argparse.ArgumentParser()
-    parser.add_argument('--path', type=str, default="static/data/lms")
-    parser.add_argument('--batch_size', type=int, default=100)
+    parser.add_argument('--specific_folder', dest='specific_folder', action='store_true', default=False)
     parser.add_argument('--delete_all', dest='delete_all', action='store_true', default=True)
+    parser.add_argument('--path', type=str, default=dirname)
+    parser.add_argument('--batch_size', type=int, default=100)
     parser.add_argument('--provider_code', type=int)
     args = parser.parse_args()
+
+    if args.specific_folder:
+        dirname = tkFileDialog.askdirectory(initialdir=os.path.abspath(dirname),  title='Please select a directory')
```

Comment:
If `--path` is given, it would make sense to use it rather than `dirname` here, no?